.github/workflows/publish.yml

```diff
@@ -0,0 +1,70 @@
+name: publish
+
+on:
+  pull_request:
+    branches: [master]
+  push:
+    branches: [master]
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: 'pages'
+  cancel-in-progress: true
+
+jobs:
+  checks:
+    if: github.event_name != 'push'
```

Comment:
If one opens a PR and pushes additional commits, will this still get triggered?

Reply:
https://github.com/orgs/community/discussions/25161 yep! It'll still get triggered.